marklogic-data-hub/src/main/java/com/marklogic/hub/deploy/commands/DeployHubDatabaseCommand.java

```diff
@@ -40,6 +40,7 @@
     private File databaseFile;
     private boolean mergeEntityConfigFiles = true;
     private boolean mergeExistingArrayProperties = false;
+    private boolean removeSchemaAndTriggersDatabaseSettings = false;
```

Comment:
My analysis of 6981 was that the root cause here is that DhsDeployer should not call setKnownValuesForDhsDeployment unless mlIsProvisionedEnvironment=true. I'm not clear on what problem is being fixed here. This logic is saying "If this is DHS, then we know that including schema-database / triggers-database will result in a privilege error, so we don't include them". That seems unrelated to the problem in 6981.

Reply:
It's not just dhs, the command(hubDeployAsDeveloper) would fail in on prem environments as well with a user with "data-hub-developer". Currently, we are working around it by setting hubConfig.setIsProvisionedEnvironment(true) so we are always removing staging and triggers db settings from payload. Now if we want the command to work for both for on prem and dhs, we cannot always set hubConfig.setIsProvisionedEnvironment(true). We can set removeSchemaAndTriggersDatabaseSettings=true when running this command since we know "data-hub-developer"  would always want the staging and triggers db settings removed from payload.